tests/test_loss_mitigation.py

```diff
@@ -70,3 +75,64 @@ def test_photon_loss_mitigation(result_type):
     for keys, value in mitigated_dist.items():
         assert sum(keys) == ideal_photon_count
         assert value > 1e-6
+
+def _compute_random_circ_probs(source_emission, num_photons):
+
+    mzi = (BS() // (0, PS(phi=Parameter("φ_a")))
+           // BS() // (1, PS(phi=Parameter("φ_b"))))
+
+    random_loc = Circuit.decomposition(Matrix.random_unitary(20), mzi,
```

Comment:
Why do you need to decompose your unitary matrix into an actual interferometer?
I think you just need:
```python
random_loc = Unitary(Matrix.random_unitary(20))
``` 
It'll run a lot faster and simulate the same thing (minus the floating point inaccuracy injected by the `decomposition`).

Reply:
Yes, you are correct. I did not need it. I had it from the notebook in my scripts and missed to remove it. Fixed.